### PR TITLE
Add daily cron job to sync system address book

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Why not sync it once a user is added? Just to clarify: add a job to the cron queue like it is done for files trashbin. This would make it easier and less processing intensive (only run the job, when it is needed). Also the background job is automatically deduplicated on bulk import of users (if the job already exists it isn't added again).

**Comment by MorrisJobke** [comment]
> Just to clarify: add a job to the cron queue like it is done for files trashbin or files versions.

Referring to this: https://github.com/owncloud/core/blob/master/apps/files_versions/command/expire.php

File: apps/dav/appinfo/app.php [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 $app = new \OCA\Dav\AppInfo\Application();
 $app->registerHooks();
+$app->setupCron();
```
Comment (nickvergessen) [inline apps/dav/appinfo/app.php]: This should be in install/update so we don't execute the "does job already exist" query on every page load?

**Comment by DeepDiver1975** [comment]
> Why not sync it once a user is added? Just to clarify: add a job to the cron queue like it is done for files trashbin or files versions. This would make it easier and less processing intensive (only run the job, when it is needed). Also the background job is automatically deduplicated on bulk import of users (if the job already exists it isn't added again).

Generally speaking: we update the system addressbook on every user change (there are prs pending).
Nevertheless a daily sync should keep the data sources in sync.

File: apps/dav/appinfo/app.php (hunk 1 above)
Comment (DeepDiver1975) [inline apps/dav/appinfo/app.php]: quite valid - THX

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by rullzer** [comment]
:+1:

**Pull request merged**